### PR TITLE
Add command line option to specify the wanted package format

### PR DESCRIPTION
**Pull request by ryanskeith**
### Description

Fix for issue #4890.
- Created enum class CondaPkgFormat to replace string usage of types
- `--package-format` introduced as command line argument
   - This takes precedence over default value and .condarc
   - Normalization occurs so 1,tar.bz2,.tar.bz2,2,conda,.conda are all recognized and mapped appropriately
   - Other options are rejected
   - Action created so in future, multiple outputs can be explored

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda_build/config.py [hunk 1]
```diff
@@ -58,7 +99,7 @@ def set_invocation_time():
 no_rewrite_stdout_env_default = "false"
 ignore_verify_codes_default = []
 exit_on_verify_error_default = False
-conda_pkg_format_default = None
+conda_pkg_format_default = CondaPkgFormat.V1
```
Comment (beeankha) [inline conda_build/config.py:102]:
```suggestion
conda_pkg_format_default = CondaPkgFormat.V2
```

File: news/4890-package-format-cli-options [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### Enhancements
+
+* --package-format introduced as command line argument
```
Comment (jaimergp) [inline news/4890-package-format-cli-options:3]:
```suggestion
* `--package-format` introduced as command line argument. (#4890 via #5209)
```

File: conda_build/config.py (hunk 1 above)
Comment (jaimergp) [reply conda_build/config.py:104]: I think this change should be postponed to 25.1.

**Review by jezdez — DISMISSED** [review]

File: conda_build/cli/main_build.py [hunk 3]
```diff
@@ -482,12 +487,23 @@ def parse_args(args: Sequence[str] | None) -> tuple[ArgumentParser, Namespace]:
...
+        help=(
+            "Choose which package type(s) are outputted. (Accepted inputs tar.bz2 or 1, .conda or 2)"
+        ),
+    )
     add_parser_channels(parser)
 
     parsed = parser.parse_args(args)
     check_recipe(parsed.recipe)
-    return parser, parsed
+    return parsed
```
Comment (jezdez) [inline conda_build/cli/main_build.py:506]: This seems to break a few tests that assume the parsed args are returned as well

**Review by jezdez — CHANGES_REQUESTED** [review]

File: conda_build/cli/main_build.py (hunk 3 above)
Comment (jezdez) [inline conda_build/cli/main_build.py:506]: This seems to break a few tests that assume the parsed args are returned as well

```suggestion
    return parser, parsed
```

File: conda_build/cli/main_build.py [hunk 4]
```diff
@@ -482,7 +487,18 @@ def parse_args(args: Sequence[str] | None) -> tuple[ArgumentParser, Namespace]:
...
+    parser.add_argument(
+        "--package-format",
+        dest="conda_pkg_format",
+        choices=CondaPkgFormat.acceptable(),
+        action=PackageTypeNormalize,
+        default=CondaPkgFormat.normalize(
+            context.conda_build.get("pkg_format", conda_pkg_format_default)
+        ),
+        help=(
+            "Choose which package type(s) are outputted. (Accepted inputs tar.bz2 or 1, .conda or 2)"
```
Comment (dholth) [inline conda_build/cli/main_build.py:499]:
```suggestion
            "Choose which package type(s) are outputted. (Accepted inputs .tar.bz2 or 1, .conda or 2)"
```

File: conda_build/cli/main_build.py [hunk 5]
```diff
@@ -85,7 +90,7 @@ def parse_args(args: Sequence[str] | None) -> tuple[ArgumentParser, Namespace]:
         "-t",
         "--test",
         action="store_true",
-        help="Test package (assumes package is already built).  RECIPE_DIR argument must be a "
+        help="Test package (assumes package is already built).  RECIPE_PATH argument must be a "
         "path to built package .tar.bz2 file.",
```
Comment (dholth) [inline conda_build/cli/main_build.py:94]:
```suggestion
        "path to built package file.",
```
Does it only support tarbz2

**Comment by dholth** [comment]
Isn't there also an environment variable for this

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**